### PR TITLE
[TS migration] Migrate 'SelectionList' component to TypeScript

### PR DESCRIPTION
**Pull request by filip-solecki**
### Fixed Issues

$ https://github.com/Expensify/App/issues/31981
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

(SelectionList with RadioItem)
1. Log In to app
2. Go to Settings
3. Go to Profile
4. Go to Timezone
5. Select Timezone field
6. Test SelectionList

(SelectionList with UserItem)
1. Log In to app
2. Go to Settings
3. Go to Workspaces
4. Select Workspace
5. Go to Members
6. Test SelectionList

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/153545991/0311e1cf-19d1-4ef8-94fa-78224ed232cc

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/153545991/f92afed8-b35a-42af-8282-6f7f9621b023

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/153545991/5f63fe1b-fe81-40fc-88d4-92db03dd8e16

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/153545991/ab5f7b3d-a121-47e0-98e5-6f8d7d0a1da6

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/153545991/8e119d45-d0bf-430d-a3ec-07b5201b56e4

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/153545991/e742f94a-bee2-40db-9886-8e7f5627a6a5

</details>

### CODE REVIEW

File: src/components/SelectionList/index.tsx [hunk 1]
```diff
@@ -14,8 +17,6 @@ const SelectionList = forwardRef((props, ref) => {
             return;
         }
 
-        // We're setting `isScreenTouched` in this listener only for web platforms with touchscreen (mWeb) where
```
Comment (blazejkustra) [inline src/components/SelectionList/index.tsx:17]: Why these comments were removed?

File: src/components/SelectionList/index.tsx [hunk 2]
```diff
@@ -31,16 +32,15 @@ const SelectionList = forwardRef((props, ref) => {
...
                 }
                 Keyboard.dismiss();
             }}
         />
     );
-});
+}
 
-SelectionList.displayName = 'SelectionList';
+SelectionList.displayName = 'SelectionListFunction';
```
Comment (blazejkustra) [inline src/components/SelectionList/index.tsx:44]:
```suggestion
SelectionList.displayName = 'SelectionList';
```

File: src/components/SelectionList/UserListItem.tsx [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
-import {userListItemPropTypes} from './selectionListPropTypes';
+import type {UserListItemProps} from './types';
 
-function UserListItem({item, textStyles, alternateTextStyles, showTooltip}) {
+function UserListItem({item, textStyles, alternateTextStyles, showTooltip}: UserListItemProps) {
     const styles = useThemeStyles();
     return (
         <>
-            {Boolean(item.icons) && (
+            {item.icons && item.icons?.length > 1 && (
```
Comment (blazejkustra) [inline src/components/SelectionList/UserListItem.tsx:13]:
```suggestion
            {!!item.icons?.length && (
```

File: src/components/SelectionList/BaseListItem.tsx [hunk 4]
```diff
@@ -111,7 +120,7 @@ function BaseListItem({
                         </View>
                     )}
                 </View>
-                {Boolean(item.invitedSecondaryLogin) && (
+                {item.invitedSecondaryLogin && (
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseListItem.tsx:123]:
```suggestion
                {!!item.invitedSecondaryLogin && (
```

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/SelectionList/index.tsx (hunk 2 above)
Comment (filip-solecki) [reply src/components/SelectionList/index.tsx]: Done

File: src/components/SelectionList/BaseListItem.tsx (hunk 4 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseListItem.tsx]: Done

File: src/components/SelectionList/BaseListItem.tsx [hunk 5]
```diff
@@ -42,14 +40,14 @@ function BaseListItem({
         >
             <PressableWithFeedback
                 onPress={() => onSelectRow(item)}
-                disabled={isDisabled}
+                disabled={item.isDisabled}
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseListItem.tsx:43]: Why was this changed?

File: src/components/SelectionList/BaseListItem.tsx [hunk 6]
```diff
@@ -42,14 +40,14 @@ function BaseListItem({
...
                 accessibilityLabel={item.text}
                 role={CONST.ROLE.BUTTON}
                 hoverDimmingValue={1}
                 hoverStyle={styles.hoveredComponentBG}
                 dataSet={{[CONST.SELECTION_SCRAPER_HIDDEN_ELEMENT]: true}}
                 onMouseDown={shouldPreventDefaultFocusOnSelectRow ? (e) => e.preventDefault() : undefined}
-                nativeID={keyForList}
+                testID={keyForList}
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseListItem.tsx:50]: Why was this changed?

File: src/components/SelectionList/BaseListItem.tsx [hunk 7]
```diff
@@ -84,20 +82,38 @@ function BaseListItem({
...
-                            styles.pre,
-                            item.alternateText ? styles.mb1 : null,
-                        ]}
-                        alternateTextStyles={[styles.optionAlternateText, styles.textLabelSupporting, isFocused ? styles.sidebarLinkActiveText : styles.sidebarLinkText, styles.pre]}
-                        isDisabled={isDisabled}
-                        onSelectRow={onSelectRow}
-                        showTooltip={showTooltip}
-                    />
+
+                    {item.rightElement === undefined ? (
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseListItem.tsx:86]: Is this necessary? It looked, cleaner before

File: src/components/SelectionList/UserListItem.tsx [hunk 8]
```diff
@@ -1,20 +1,19 @@
...
-import {userListItemPropTypes} from './selectionListPropTypes';
+import type {UserListItemProps} from './types';
 
-function UserListItem({item, textStyles, alternateTextStyles, showTooltip}) {
+function UserListItem({item, textStyles, alternateTextStyles, showTooltip}: UserListItemProps) {
     const styles = useThemeStyles();
     return (
         <>
-            {Boolean(item.icons) && (
+            {!!item.icons && item.icons?.length > 1 && (
```
Comment (blazejkustra) [inline src/components/SelectionList/UserListItem.tsx:13]: Why not just:
```suggestion
            {!!item.icons && (
```

File: src/components/SelectionList/types.ts [hunk 9]
```diff
@@ -0,0 +1,241 @@
+import type {ReactNode} from 'react';
+import type {GestureResponderEvent, InputModeOptions, StyleProp, TextStyle, ViewStyle} from 'react-native';
+import type {SubAvatar} from '@components/SubscriptAvatar';
+import type {Errors, PendingAction} from '@src/types/onyx/OnyxCommon';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+
+type CommonListItemProps<T> = {
```
Comment (blazejkustra) [inline src/components/SelectionList/types.ts:7]:
```suggestion
type CommonListItemProps<TItem> = {
```

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/SelectionList/BaseListItem.tsx (hunk 6 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseListItem.tsx:50]: It was done by Kuba Szymczak so I have no idea tbh, should I revert this change?

Comment (blazejkustra) [reply src/components/SelectionList/BaseListItem.tsx:50]: Yes 😄

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 10]
```diff
@@ -230,15 +219,16 @@ function BaseSelectionList({
     };
 
     const selectAllRow = () => {
-        onSelectAll();
+        if (onSelectAll) {
+            onSelectAll();
+        }
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseSelectionList.tsx:224]:
```suggestion
        onSelectAll?.();
```

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 11]
```diff
@@ -230,15 +219,16 @@ function BaseSelectionList({
...
         if (shouldShowTextInput && shouldPreventDefaultFocusOnSelectRow && textInputRef.current) {
             textInputRef.current.focus();
         }
     };
 
-    const selectFocusedOption = (e) => {
-        const focusedItemKey = lodashGet(e, ['target', 'attributes', 'id', 'value']);
-        const focusedOption = focusedItemKey ? _.find(flattenedSections.allOptions, (option) => option.keyForList === focusedItemKey) : flattenedSections.allOptions[focusedIndex];
+    const selectFocusedOption = () => {
+        const focusedOption = flattenedSections.allOptions[focusedIndex];
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseSelectionList.tsx:231]: Why was this removed? 
```ts
       const focusedItemKey = lodashGet(e, ['target', 'attributes', 'id', 'value']);
        const focusedOption = focusedItemKey ? _.find(flattenedSections.allOptions, (option) => option.keyForList === focusedItemKey) : flattenedSections.allOptions[focusedIndex];
```

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 12]
```diff
@@ -179,7 +168,7 @@ function BaseSelectionList({
             // Otherwise, it will cause an index-out-of-bounds error and crash the app.
             let adjustedSectionIndex = sectionIndex;
             for (let i = 0; i < sectionIndex; i++) {
-                if (_.isEmpty(lodashGet(sections, `[${i}].data`))) {
+                if (sections[i].data) {
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseSelectionList.tsx:171]: Is optional chain needed here?
```suggestion
                if (sections?.[i]?.data) {
```

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 13]
```diff
@@ -469,7 +467,7 @@ function BaseSelectionList({
                                     onScrollBeginDrag={onScrollBeginDrag}
                                     keyExtractor={(item) => item.keyForList}
                                     extraData={focusedIndex}
-                                    indicatorStyle={theme.white}
+                                    indicatorStyle="white"
```
Comment (blazejkustra) [inline src/components/SelectionList/BaseSelectionList.tsx:470]: Why was this changed?

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 11 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx:231]: It was always undefined as property id does not exist on attributes

File: src/components/SelectionList/BaseListItem.tsx (hunk 7 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseListItem.tsx:86]: Unfortunately there is no way to make it shorter e.g. as it was before because of type difference between those components

File: src/components/SelectionList/BaseListItem.tsx (hunk 5 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseListItem.tsx]: Reverted

File: src/components/SelectionList/BaseListItem.tsx (hunk 6 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseListItem.tsx:50]: Reverted

File: src/components/SelectionList/UserListItem.tsx (hunk 8 above)
Comment (filip-solecki) [reply src/components/SelectionList/UserListItem.tsx]: Done

File: src/components/SelectionList/types.ts (hunk 9 above)
Comment (filip-solecki) [reply src/components/SelectionList/types.ts]: Renamed

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 14]
```diff
@@ -426,12 +424,12 @@ function BaseSelectionList({
...
                             </View>
                         )}
-                        {Boolean(headerMessage) && (
+                        {headerMessage && (
                             <View style={[styles.ph5, styles.pb5]}>
                                 <Text style={[styles.textLabel, styles.colorMuted]}>{headerMessage}</Text>
                             </View>
                         )}
-                        {Boolean(headerContent) && headerContent}
+                        {headerContent}
```
[the remark this answers is not in the document]
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx]: Done

File: src/components/SelectionList/types.ts [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+    isDisabled?: boolean;
+
+    /** User accountID */
+    accountID?: number;
+
+    /** User login */
+    login?: string;
+
+    /** Element to show on the right side of the item */
+    rightElement: JSX.Element;
```
Comment (blazejkustra) [inline src/components/SelectionList/types.ts:56]: Can it be ReactNode?
```suggestion
    rightElement: ReactNode
```

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 13 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx:470]: indicatorStyle accepts only: `'default' | 'black' | 'white' | undefined;` and theme.white is just string

**Review by blazejkustra — APPROVED** [review]

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 16]
```diff
@@ -112,7 +101,7 @@ function BaseSelectionList({
                 });
 
                 // If disabled, add to the disabled indexes array
-                if (section.isDisabled || item.isDisabled) {
+                if (section.isDisabled ?? item.isDisabled) {
```
[the remark this answers is not in the document]
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx]: Yep

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 10 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx:224]: Done

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 12 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx:171]: No need - sections is required

File: src/components/SelectionList/types.ts (hunk 15 above)
Comment (filip-solecki) [reply src/components/SelectionList/types.ts:56]: Will be ReactElement

**Comment by blazejkustra** [comment]
@filip-solecki conflicts

File: src/components/SelectionList/BaseListItem.tsx [hunk 17]
```diff
@@ -25,13 +24,12 @@ function BaseListItem({
...
     keyForList,
-}) {
+}: BaseListItemProps) {
     const theme = useTheme();
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {translate} = useLocalize();
-    const isUserItem = lodashGet(item, 'icons.length', 0) > 0;
-    const ListItem = isUserItem ? UserListItem : RadioListItem;
+    const isRadioItem = item.rightElement === undefined;
```
Comment (kubabutkiewicz) [inline src/components/SelectionList/BaseListItem.tsx:32]: why this is different from original?

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 18]
```diff
@@ -252,19 +241,17 @@ function BaseSelectionList({
...
      *     2. Each section includes a header, even if we don't provide/render one.
      *
      *     For example, given a list with two sections, two items in each section, no header, no footer, and no section headers, the flat array might look something like this:
      *
      *     [{header}, {sectionHeader}, {item}, {item}, {sectionHeader}, {item}, {item}, {footer}]
-     *
-     * @returns {Object}
      */
-    const getItemLayout = (data, flatDataArrayIndex) => {
+    const getItemLayout = (data: Array<SectionListData<User | RadioItem, Section>> | null, flatDataArrayIndex: number) => {
```
Comment (kubabutkiewicz) [inline src/components/SelectionList/BaseSelectionList.tsx:254]: `SectionListData<User | RadioItem, Section>` can be separate type to reuse in different places

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 19]
```diff
@@ -310,11 +298,9 @@ function BaseSelectionList({
                 item={item}
                 isFocused={isItemFocused}
                 isDisabled={isDisabled}
-                isHide={!maxToRenderPerBatch}
```
Comment (kubabutkiewicz) [inline src/components/SelectionList/BaseSelectionList.tsx:313]: why props got removed from here ?

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 20]
```diff
@@ -398,19 +388,26 @@ function BaseSelectionList({
...
+                    <View style={[styles.flex1, !isKeyboardShown && safeAreaPaddingBottomStyle, containerStyle]}>
                         {shouldShowTextInput && (
                             <View style={[styles.ph5, styles.pb3]}>
                                 <TextInput
-                                    ref={(el) => {
-                                        if (inputRef) {
-                                            // eslint-disable-next-line no-param-reassign
-                                            inputRef.current = el;
+                                    ref={(element) => {
+                                        textInputRef.current = element as RNTextInput;
```
Comment (kubabutkiewicz) [inline src/components/SelectionList/BaseSelectionList.tsx:398]: why this was added?

**Review by kubabutkiewicz — COMMENTED** [review]
left few comments , also typecheck is failing

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 19 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx:313]: Not used in BaseListItem component

File: src/components/SelectionList/BaseListItem.tsx (hunk 17 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseListItem.tsx:32]: It is simplier and easier to differ types `User` and `RadioItem` as `rightElement` is required for `User` type

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 20 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx]: You're right - it is not needed.
Removed

File: src/components/SelectionList/BaseSelectionList.tsx (hunk 18 above)
Comment (filip-solecki) [reply src/components/SelectionList/BaseSelectionList.tsx]: Done

**Comment by filip-solecki** [comment]
@kubabutkiewicz Back to you!

**Review by kubabutkiewicz — APPROVED** [review]

**Comment by allroundexperts** [comment]
@filip-solecki Conflicts!

**Comment by filip-solecki** [comment]
@allroundexperts confilcts resolved!

**Comment by allroundexperts** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by allroundexperts — APPROVED** [review]
Looks good!

File: src/components/SelectionList/types.ts [hunk 21]
```diff
@@ -0,0 +1,273 @@
...
+    errors?: Errors;
+
+    /** The type of action that's pending  */
+    pendingAction?: PendingAction;
+
+    invitedSecondaryLogin?: string;
+
+    sectionIndex: number; // mb throw this out
+
+    index: number; // mb throw this out
```
Comment (mountiny) [inline src/components/SelectionList/types.ts:74]: Should we resolve these comments?

**Review by mountiny — DISMISSED** [review]
looks good, one comment

File: src/components/SelectionList/types.ts (hunk 21 above)
Comment (filip-solecki) [reply src/components/SelectionList/types.ts]: I don't think so - it was just developer comment while working on this issue.

File: src/components/SelectionList/types.ts [hunk 22]
```diff
@@ -0,0 +1,273 @@
...
+
+    /** Errors that this user may contain */
+    errors?: undefined;
+
+    /** The type of action that's pending  */
+    pendingAction?: undefined;
+
+    sectionIndex: number; // smb throw this out
+
+    index: number; // mb throw this out
```
Comment (mountiny) [inline src/components/SelectionList/types.ts:114]: here too

**Review by mountiny — APPROVED** [review]

**Comment by Julesssss** [comment]
I'm pretty sure[ this PR](https://github.com/Expensify/App/issues/34725) introduced this regression. Are you able to submit a fix today? If not I'll revert it and we can work on an alternative fix in the meantime. Thanks

**Comment by Julesssss** [comment]
Actually I'm going to straight revert. Please prepare another PR that doesn't include the bug. Thanks

**Review by rushatgabhane — COMMENTED** [review]
Small issue that was not handled from https://github.com/Expensify/App/issues/42207,
